packages/smart-contracts/src/contracts/ERC20SwapToPay.sol

```diff
@@ -1,8 +1,9 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.0;
+pragma solidity 0.8.4;
 
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
+import "./legacy_openzeppelin/contracts/access/roles/WhitelistedRole.sol";
```

Comment:
Since you're using `solidity 0.8.4`, can't you use one of the non-legacy OZ contracts?